packages/cli/mock/src/runMockServer.ts

```diff
@@ -230,6 +231,35 @@ function getRequestHandler(endpoints: HttpEndpoint[]): RequestHandler {
     };
 }
 
+function isValidDate(value: string): boolean {
+    const date = new Date(value);
+    return !isNaN(date.getTime());
+}
+
+// Customize the lodash isEqualWith function to handle date-like strings
+// returning undefined in this comparitor indicates to lodash to perform it's
+// internal comparison logic.
+function dateishCustomizer(value: unknown, other: unknown): boolean | undefined {
+    if (isPlainObject(value) || isPlainObject(other)) {
+        return undefined;
+    }
+    if (typeof value === "string" && typeof other === "string") {
+        // If the strings are equal, short circuit and return true
+        if (value === other) {
+            return true;
+        }
+
+        // Otherwise check if they're dates to give them another chance
+        if (isValidDate(value) && isValidDate(other)) {
+            const valueDate = new Date(value);
+            const otherDate = new Date(other);
+            // Is variance less than a day
+            return Math.abs(valueDate.getTime() - otherDate.getTime()) <= 24 * 60 * 60 * 1000;
+        }
+    }
+    return undefined;
+}
```

Comment:
@armandobelardo can we add a quick test to ensure that the mock server works as expected?